app/org/sagebionetworks/bridge/dynamodb/DynamoCompoundActivityDefinitionDao.java

```diff
@@ -0,0 +1,133 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Resource;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.model.ConditionalCheckFailedException;
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.dao.CompoundActivityDefinitionDao;
+import org.sagebionetworks.bridge.exceptions.ConcurrentModificationException;
+import org.sagebionetworks.bridge.exceptions.EntityNotFoundException;
+import org.sagebionetworks.bridge.models.schedules.CompoundActivityDefinition;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
+
+/** DynamoDB implementation of CompoundActivityDefinitionDao. */
+@Component
+public class DynamoCompoundActivityDefinitionDao implements CompoundActivityDefinitionDao {
+    private DynamoDBMapper mapper;
+
+    /** DDB mapper, configured by Spring. */
+    @Resource(name = "compoundActivityDefinitionDdbMapper")
+    public final void setDdbMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public CompoundActivityDefinition createCompoundActivityDefinition(StudyIdentifier studyId,
+            CompoundActivityDefinition compoundActivityDefinition) {
+        // Currently, all CompoundActivityDefinitions are DynamoCompoundActivityDefinitions.
+        DynamoCompoundActivityDefinition ddbDef = (DynamoCompoundActivityDefinition) compoundActivityDefinition;
+
+        // Set study to prevent people from creating defs in other studies.
+        ddbDef.setStudyId(studyId.getIdentifier());
```

Comment:
I would do this in the service, so if we move off of DDB, all this logic is preserved in the re-usable service. If I could do it over, I'd put the DAO in a package with the service, package visible, so only the service can use it, and no one can side-step the logic in the service.